### PR TITLE
Fix EZP-23984: Refactor: FieldRegistry should be shared by all search engines

### PR DESCRIPTION
**Pull request by pspanja**
This PR resolves https://jira.ez.no/browse/EZP-23984, sub-task of https://jira.ez.no/browse/EZP-23940

Followup: https://jira.ez.no/browse/EZP-23985

This moves FieldRegistry into a common namespace and adapts the configuration, making it shared by all search engines (bar Legacy).

Symfony integration was manually tested.

### CODE REVIEW

**Comment by pspanja** [comment]
Ready for review, ping @andrerom, @bdunogier, @glye, @lolautruche, @yannickroger.

**Comment by glye** [comment]
+1, though it's more kosher to squash those 15 commits, isn't it?

**Comment by pspanja** [comment]
@glye normally yes, but I did that on purpose as it needs to be backported to 5.4 and 5.3.

File: eZ/Publish/Core/Search/Solr/Content/FieldMap.php [hunk 1]
```diff
@@ -13,6 +13,7 @@
 use eZ\Publish\API\Repository\Values\Content\Query\CustomFieldInterface;
 use eZ\Publish\API\Repository\Values\Content\Query\SortClause;
 use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\Core\Search\Common\FieldRegistry;
```
Comment (lolautruche) [inline eZ/Publish/Core/Search/Solr/Content/FieldMap.php]: It seems this import is not needed

File: eZ/Publish/Core/Search/Solr/Content/Handler.php [hunk 2]
```diff
@@ -23,6 +23,7 @@
 use eZ\Publish\SPI\Search\FieldType;
 use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
 use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\Core\Search\Common\FieldRegistry;
```
Comment (lolautruche) [inline eZ/Publish/Core/Search/Solr/Content/Handler.php]: Not needed

**Comment by lolautruche** [comment]
+1

> @glye normally yes, but I did that on purpose as it needs to be backported to 5.4 and 5.3.

I don't get it. All commits should be squashed *especially* for this very reason

File: eZ/Publish/Core/Search/Solr/Content/FieldMap.php (hunk 1 above)
Comment (pspanja) [inline eZ/Publish/Core/Search/Solr/Content/FieldMap.php]: Yes, it is a service dependency here.

File: eZ/Publish/Core/Search/Solr/Content/Handler.php (hunk 2 above)
Comment (pspanja) [inline eZ/Publish/Core/Search/Solr/Content/Handler.php]: Yes, it is a service dependency here.

**Comment by pspanja** [comment]
@lolautruche 5.4 will get Elasticsearch and Solr, while 5.3 has only Solr. Also I don't know what kind of conflicts I can expect down there, so hoping it will make my life a bit easier. Backports will of course be squashed into one commit.

**Comment by pspanja** [comment]
Thanks for the review!

**Pull request merged**